File: segmentSupport/src/main/java/com/clumob/segment/support/pager/SegmentItemProvider.java
package com.clumob.segment.support.pager;

import com.clumob.listitem.controller.source.ItemController;
import com.clumob.segment.controller.SegmentController;
import com.clumob.segment.controller.SegmentInfo;
import com.clumob.segment.manager.Segment;

/**
 * Created by prashant.rathore on 02/07/18.
 */

public interface SegmentItemProvider<IC extends ItemController, VM, SC extends SegmentController<VM>> {
    public Segment<VM,SC> provide(IC segmentInfo);
}

File: segmentSupport/src/main/java/com/clumob/segment/support/pager/SegmentItemProvider.kt
package com.clumob.segment.support.pager

import com.clumob.segment.controller.common.Controller
import com.clumob.segment.manager.Segment

/**
 * Created by prashant.rathore on 02/07/18.
 */
interface SegmentItemProvider {
    fun provide(segmentItem: Controller): Segment
}